geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionImpl.java

```diff
@@ -94,9 +99,10 @@ public ConnectionImpl(InternalDistributedSystem ds, CancelCriterion cancelCriter
   }
 
   public static int getClientFunctionTimeout() {
```

Comment:
Why is these even a thing on the connection? Things that are done with a connection, like function execution, shouldn't be leaking into the API of connection. Perhaps this needs to move out of here completely.

Reply:
I agree that this does not make sense on ConnectionImpl.
How about moving it to: InternalFunctionExecutionServiceImpl. You could make it a non-static final instance variable on this class. Change to code that want to read this timeout to use InternalFunctionService.getInternalFunctionExecutionService() to get a reference to the InternalFunctionExecutionServiceImpl. I currently don't know if a way to get a reference to the function service other than this static method. 
Also the three places that call getClientFunctionTimeout all compare it to DEFAULT_CLIENT_FUNCTION_TIMEOUT and only add byte part if it is not the default. I think this code would be a little bit easier to understand if it just compared it to 0. We only want to add the byte part if it is not zero. If someone changed the default in the future we would still only want to add the byte part if it is not zero. We don't care in that code if the timeout is the default. What we care about is if it is zero or not.

Reply:
I am going to roll back my request. This thing is broken on even more levels than I thought. Notice that `ConnectionImpl.execute(Op op)` uses it if the `op` is of function variants. So even if we moved the method or where we read this system property we still have leakage into `ConnectionImpl`. I will open a ticket to address this leakage completely.

I now think you should do what @kirklund suggested and make the constructor of `ConnectionImpl` read the value and stash it `final`.

Reply:
I agree with Jake. Just make it a non-static final.

Reply:
Just to clarify, should we just make it non-static final class variable, or make it final instance variable which will be initialized in constructor.

Reply:
Final instance variable and init in constructor.